Jenkinsfile

```diff
@@ -25,12 +25,23 @@ pipeline {
         TIMEOUT             = '120m'
     }
     stages {
+        stage('pr-hack') {
+            when { changeRequest() }
+            steps {
+                script {
+                    echo "Workaround for PR auto-cancel feature. Borrowed from https://issues.jenkins-ci.org/browse/JENKINS-43353"
+                    def buildNumber = env.BUILD_NUMBER as int
+                    if (buildNumber > 1) milestone(buildNumber - 1)
+                    milestone(buildNumber)
+                }
+            }
+        }
         stage('DCO-check') {
             when {
                 beforeAgent true
                 expression { !params.skip_dco }
             }
-            agent { label 'linux' }
+            agent { label 'amd64 && ubuntu-1804 && overlay2' }
```

Comment:
Wondering; any advantage to being more specific here? (I left it generic, thinking "any Linux node will do, so give me whatever you have available")

Reply:
Any "linux" node will work in most cases as you would expect. I believe it is better to be more specific to avoid the "wrong node type" class of problems that could creep up later.